sys/include/link_format.h

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (C) 2019 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_link_format Link Format
+ * @ingroup     sys_serialization
+ * @brief       Simple encoding and decoding of Link Format strings
+ *
+ * @{
+ *
+ * @file
+ * @brief       Link format encoding and decoding library public definitions
+ *
+ * @author      Leandro Lanzieri <leandro.lanzieri@haw-hamburg.de>
+ */
+
+#ifndef LINK_FORMAT_H
+#define LINK_FORMAT_H
+
+#include <sys/types.h>
+
+#include "link_format_internal.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Return types for the @ref sys_link_format API
+ */
+enum {
+    LINK_FORMAT_OK          = 0,    /**< success */
+    LINK_FORMAT_NO_SPACE    = -1,   /**< not enough space in the buffer */
+    LINK_FORMAT_NOT_FOUND   = -2    /**< could not find a component in a
+                                         buffer */
+};
+
+/**
+ * @brief Types of link format parameters
+ */
+typedef enum {
+    LINK_FORMAT_PARAM_ANCHOR        = 0,  /**< anchor */
+    LINK_FORMAT_PARAM_REL_TYPE      = 1,  /**< rel */
+    LINK_FORMAT_PARAM_LANG          = 2,  /**< hreflang */
+    LINK_FORMAT_PARAM_MEDIA         = 3,  /**< media */
+    LINK_FORMAT_PARAM_TITLE         = 4,  /**< title */
+    LINK_FORMAT_PARAM_TITLE_EXT     = 5,  /**< title* */
+    LINK_FORMAT_PARAM_TYPE          = 6,  /**< type */
+    LINK_FORMAT_PARAM_RES_TYPE      = 7,  /**< rt */
```

Comment:
also applies for other fields: I think keeping the short form, e.g. `rt` as part of the enum member name is likely more straight forward for users, right?

Reply:
Yes, I shortened them